browser/ui/views/download/brave_download_item_view.cc

```diff
@@ -236,18 +236,32 @@ void BraveDownloadItemView::SetMode(download::DownloadItemMode mode) {
   }
 }
 
+void BraveDownloadItemView::UpdateLabels() {
+  DownloadItemView::UpdateLabels();
+  // Update visibility to avoid artifacts to be shown from upstream
+  file_name_label_->SetVisible(
+      !is_origin_url_visible_ &&
+      (GetMode() == download::DownloadItemMode::kNormal));
+}
+
+void BraveDownloadItemView::SetOriginUrlVisible(bool visible) {
+  is_origin_url_visible_ = visible;
+  file_name_label_->SetVisible(
+      !visible && (GetMode() == download::DownloadItemMode::kNormal));
+}
```

Comment:
If we're overriding `UpdateLabels` in order to react to the change in property and call SetVisible accordingly, then we probably don't need to duplicate the call to `file_name_label_->SetVisible` here, we just need to make sure UpdateLabels gets called, and paint is invalidated so that our origin gets painted. I think we can just instead call `OnDownloadUpdated()` which would call `SetMode` and run all that.

Reply:
This works fine. However, the browser crashes on an equality check made inside parent's `OnPaint` when reproducing the issue the above ticket addresses. 

Turn your network off, click "resume" which shows failed state and crashes at the following line

> [35253:259:0818/102420.760510:FATAL:download_item_view.cc(602)] Check failed: download::DownloadItemMode::kNormal == mode_ (0 vs. 4)

I've attempted to call `OnDownloadUpdated` in `SetOriginUrlVisible` based on a check if the mode is normal, it still crashes.

Reply:
I understand why that crashed, so this should work fine locally for me. Can you change to it and let me know you don't see any issues? I tested locally and looked good to me.

```cpp
void BraveDownloadItemView::SetOriginUrlVisible(bool visible) {
  is_origin_url_visible_ = visible;
  UpdateLabels();
}
```

Reply:
this works!